### PR TITLE
Pseudolimbs refactor

### PR DESCRIPTION
**Pull request by QualityVan**
:cl: QV
tweak: Refactored the way limbs that aren't limbs work
/:cl:

Adds a flag for limbs that don't really exist as in the case of attached chainsaws. Prevents doing organ manipulation or augmentation on chainsaws, and stops an arm from being dropped when a chainsaw is dismembered off someone.

### CODE REVIEW

**Comment by coiax** [comment]
I feel like we should revisit this, and introduce "hand" limbs. Then chainsaws and armblades can just be replaced hands.

**Comment by XDTM** [comment]
Hand limbs are simulated by NODROP items, like when you use arm-implanted tools or weapons. Chainsaws and armblades pretty much replace the whole arm, though, and it makes sense you can't augment what's not there.

**Comment by QualityVan** [comment]
Yeah, the thing about nodrop is that stuff like the defib and botany/janitor spraypacks also work off it, along with nodrop dropdel things like zombie claws/godhand. A check for nodrop and abstract and not dropdel would work for the purpose on hand, but it didn't seem as good.

File: code/modules/surgery/bodyparts/bodyparts.dm [hunk 1]
```diff
@@ -20,6 +20,7 @@
 	var/max_damage = 0
 	var/list/embedded_objects = list()
 	var/held_index = 0 //are we a hand? if so, which one!
+	var/is_psuedopart = 0 //For limbs that don't really exist, eg chainsaws
```
Comment (AnturK) [inline code/modules/surgery/bodyparts/bodyparts.dm]: Spelling and TRUE/FALSE defines please.

File: code/modules/surgery/bodyparts/bodyparts.dm [hunk 2]
```diff
@@ -195,6 +196,9 @@
 		owner.update_hair()
 		owner.update_damage_overlays()
 
+obj/item/bodypart/proc/set_fakeness(fakeness)
```
Comment (AnturK) [inline code/modules/surgery/bodyparts/bodyparts.dm]: Setter doesn't seem necessary here. Or at least i can't think of a scenario where it should have side effects.

**Comment by lzimann** [comment]
unless i dont know english, isn't it spelld "pseudo"

**Comment by RemieRichards** [comment]
it is pseudo yes

**Comment by BeeSting12** [comment]
you dont know english     😄

**Pull request merged**